src/blueprint/links.rs

```diff
@@ -0,0 +1,37 @@
+use crate::config::Link;
+use crate::directive::DirectiveCodec;
+use crate::valid::{Valid, ValidationError, Validator};
+
+// Wrap the Vec<Link> is necessary to implement TryFrom
+#[allow(dead_code)]
+pub struct Links(Vec<Link>);
```

Comment:
Creating this was necessary as `TryFrom` can't be directly implemented for `Vec` as both the trait and the struct are external types.